### PR TITLE
upgrade mssql version to 12.6.1

### LINKED ISSUE #113: investigate mssql jdbc jar upgrade
Tried using latest mssql jar but got errors. We are testing with an old jar and we should really be able to test successfully with newer jars.

https://github.com/apache/incubator-pekko-persistence-jdbc/actions/runs/7862784119

```
[info]   Cause: com.microsoft.sqlserver.jdbc.SQLServerException: "encrypt" property is set to "true" and "trustServerCertificate" property is set to "false" but the driver could not establish a secure connection to SQL Server by using Secure Sockets Layer (SSL) encryption: Error: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target. ClientConnectionId:ac43e56d-18b4-4434-9dc8-2b1cb114e7fa
[info]   at com.microsoft.sqlserver.jdbc.SQLServerConnection.terminate(SQLServerConnection.java:4266)
[info]   at com.microsoft.sqlserver.jdbc.TDSChannel.enableSSL(IOBuffer.java:1965)
[info]   at com.microsoft.sqlserver.jdbc.SQLServerConnection.connectHelper(SQLServerConnection.java:3792)
[info]   at com.microsoft.sqlserver.jdbc.SQLServerConnection.login(SQLServerConnection.java:3348)
[info]   at com.microsoft.sqlserver.jdbc.SQLServerConnection.connectInternal(SQLServerConnection.java:3179)
[info]   at com.microsoft.sqlserver.jdbc.SQLServerConnection.connect(SQLServerConnection.java:1953)
[info]   at com.microsoft.sqlserver.jdbc.SQLServerDriver.connect(SQLServerDriver.java:1263)
[info]   at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:136)
[info]   at com.zaxxer.hikari.pool.PoolBase.newConnection(PoolBase.java:369)
[info]   at com.zaxxer.hikari.pool.PoolBase.newPoolEntry(PoolBase.java:198)
[info]   at com.zaxxer.hikari.pool.HikariPool.createPoolEntry(HikariPool.java:467)
[info]   at com.zaxxer.hikari.pool.HikariPool.access$100(HikariPool.java:71)
[info]   at com.zaxxer.hikari.pool.HikariPool$PoolEntryCreator.call(HikariPool.java:706)
[info]   at com.zaxxer.hikari.pool.HikariPool$PoolEntryCreator.call(HikariPool.java:692)
[info]   at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[info]   at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[info]   at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[info]   at java.base/java.lang.Thread.run(Thread.java:829)
[info]   Cause: javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
[info]   at java.base/sun.security.ssl.Alert.createSSLException(Alert.java:131)
[info]   at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:360)
[info]   at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:303)
[info]   at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:298)
[info]   at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.checkServerCerts(CertificateMessage.java:654)
[info]   at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.onCertificate(CertificateMessage.java:473)
[info]   at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.consume(CertificateMessage.java:369)
[info]   at java.base/sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:392)
[info]   at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:443)
[info]   at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:421)
[info]   at java.base/sun.security.ssl.TransportContext.dispatch(TransportContext.java:183)
[info]   at java.base/sun.security.ssl.SSLTransport.decode(SSLTransport.java:172)
[info]   at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1511)
[info]   at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1421)
[info]   at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:456)
[info]   at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:427)
[info]   at com.microsoft.sqlserver.jdbc.TDSChannel.enableSSL(IOBuffer.java:1854)
[info]   at com.microsoft.sqlserver.jdbc.SQLServerConnection.connectHelper(SQLServerConnection.java:3792)
[info]   at com.microsoft.sqlserver.jdbc.SQLServerConnection.login(SQLServerConnection.java:3348)
[info]   at com.microsoft.sqlserver.jdbc.SQLServerConnection.connectInternal(SQLServerConnection.java:3179)
[info]   at com.microsoft.sqlserver.jdbc.SQLServerConnection.connect(SQLServerConnection.java:1953)
[info]   at com.microsoft.sqlserver.jdbc.SQLServerDriver.connect(SQLServerDriver.java:1263)
[info]   at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:136)
[info]   at com.zaxxer.hikari.pool.PoolBase.newConnection(PoolBase.java:369)
[info]   at com.zaxxer.hikari.pool.PoolBase.newPoolEntry(PoolBase.java:198)
[info]   at com.zaxxer.hikari.pool.HikariPool.createPoolEntry(HikariPool.java:467)
[info]   at com.zaxxer.hikari.pool.HikariPool.access$100(HikariPool.java:71)
[info]   at com.zaxxer.hikari.pool.HikariPool$PoolEntryCreator.call(HikariPool.java:706)
[info]   at com.zaxxer.hikari.pool.HikariPool$PoolEntryCreator.call(HikariPool.java:692)
[info]   at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[info]   at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[info]   at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[info]   at java.base/java.lang.Thread.run(Thread.java:829)
[info]   Cause: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
[info]   at java.base/sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:439)
[info]   at java.base/sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:306)
[info]   at java.base/sun.security.validator.Validator.validate(Validator.java:264)
[info]   at java.base/sun.security.ssl.X509TrustManagerImpl.validate(X509TrustManagerImpl.java:313)
[info]   at java.base/sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:233)
[info]   at java.base/sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:110)
[info]   at com.microsoft.sqlserver.jdbc.HostNameOverrideX509TrustManager.checkServerTrusted(SQLServerTrustManager.java:88)
[info]   at java.base/sun.security.ssl.AbstractTrustManagerWrapper.checkServerTrusted(SSLContextImpl.java:1492)
[info]   at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.checkServerCerts(CertificateMessage.java:638)
[info]   at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.onCertificate(CertificateMessage.java:473)
[info]   at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.consume(CertificateMessage.java:369)
[info]   at java.base/sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:392)
[info]   at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:443)
[info]   at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:421)
[info]   at java.base/sun.security.ssl.TransportContext.dispatch(TransportContext.java:183)
[info]   at java.base/sun.security.ssl.SSLTransport.decode(SSLTransport.java:172)
[info]   at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1511)
[info]   at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1421)
[info]   at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:456)
[info]   at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:427)
[info]   at com.microsoft.sqlserver.jdbc.TDSChannel.enableSSL(IOBuffer.java:1854)
[info]   at com.microsoft.sqlserver.jdbc.SQLServerConnection.connectHelper(SQLServerConnection.java:3792)
[info]   at com.microsoft.sqlserver.jdbc.SQLServerConnection.login(SQLServerConnection.java:3348)
[info]   at com.microsoft.sqlserver.jdbc.SQLServerConnection.connectInternal(SQLServerConnection.java:3179)
[info]   at com.microsoft.sqlserver.jdbc.SQLServerConnection.connect(SQLServerConnection.java:1953)
[info]   at com.microsoft.sqlserver.jdbc.SQLServerDriver.connect(SQLServerDriver.java:1263)
[info]   at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:136)
[info]   at com.zaxxer.hikari.pool.PoolBase.newConnection(PoolBase.java:369)
[info]   at com.zaxxer.hikari.pool.PoolBase.newPoolEntry(PoolBase.java:198)
[info]   at com.zaxxer.hikari.pool.HikariPool.createPoolEntry(HikariPool.java:467)
[info]   at com.zaxxer.hikari.pool.HikariPool.access$100(HikariPool.java:71)
[info]   at com.zaxxer.hikari.pool.HikariPool$PoolEntryCreator.call(HikariPool.java:706)
[info]   at com.zaxxer.hikari.pool.HikariPool$PoolEntryCreator.call(HikariPool.java:692)
[info]   at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[info]   at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[info]   at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[info]   at java.base/java.lang.Thread.run(Thread.java:829)
[info]   Cause: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
[info]   at java.base/sun.security.provider.certpath.SunCertPathBuilder.build(SunCertPathBuilder.java:148)
[info]   at java.base/sun.security.provider.certpath.SunCertPathBuilder.engineBuild(SunCertPathBuilder.java:129)
[info]   at java.base/java.security.cert.CertPathBuilder.build(CertPathBuilder.java:297)
[info]   at java.base/sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:434)
[info]   at java.base/sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:306)
[info]   at java.base/sun.security.validator.Validator.validate(Validator.java:264)
[info]   at java.base/sun.security.ssl.X509TrustManagerImpl.validate(X509TrustManagerImpl.java:313)
[info]   at java.base/sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:233)
[info]   at java.base/sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:110)
[info]   at com.microsoft.sqlserver.jdbc.HostNameOverrideX509TrustManager.checkServerTrusted(SQLServerTrustManager.java:88)
[info]   at java.base/sun.security.ssl.AbstractTrustManagerWrapper.checkServerTrusted(SSLContextImpl.java:1492)
[info]   at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.checkServerCerts(CertificateMessage.java:638)
[info]   at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.onCertificate(CertificateMessage.java:473)
[info]   at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.consume(CertificateMessage.java:369)
[info]   at java.base/sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:392)
[info]   at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:443)
[info]   at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:421)
[info]   at java.base/sun.security.ssl.TransportContext.dispatch(TransportContext.java:183)
[info]   at java.base/sun.security.ssl.SSLTransport.decode(SSLTransport.java:172)
[info]   at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1511)
[info]   at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:[1421](https://github.com/apache/incubator-pekko-persistence-jdbc/actions/runs/7862784119/job/21452649542#step:7:1422))
[info]   at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:456)
[info]   at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:427)
[info]   at com.microsoft.sqlserver.jdbc.TDSChannel.enableSSL(IOBuffer.java:1854)
[info]   at com.microsoft.sqlserver.jdbc.SQLServerConnection.connectHelper(SQLServerConnection.java:3792)
[info]   at com.microsoft.sqlserver.jdbc.SQLServerConnection.login(SQLServerConnection.java:3348)
[info]   at com.microsoft.sqlserver.jdbc.SQLServerConnection.connectInternal(SQLServerConnection.java:3179)
[info]   at com.microsoft.sqlserver.jdbc.SQLServerConnection.connect(SQLServerConnection.java:1953)
[info]   at com.microsoft.sqlserver.jdbc.SQLServerDriver.connect(SQLServerDriver.java:1263)
[info]   at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:136)
[info]   at com.zaxxer.hikari.pool.PoolBase.newConnection(PoolBase.java:369)
[info]   at com.zaxxer.hikari.pool.PoolBase.newPoolEntry(PoolBase.java:198)
[info]   at com.zaxxer.hikari.pool.HikariPool.createPoolEntry(HikariPool.java:467)
[info]   at com.zaxxer.hikari.pool.HikariPool.access$100(HikariPool.java:71)
[info]   at com.zaxxer.hikari.pool.HikariPool$PoolEntryCreator.call(HikariPool.java:706)
[info]   at com.zaxxer.hikari.pool.HikariPool$PoolEntryCreator.call(HikariPool.java:692)
[info]   at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[info]   at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[info]   at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[info]   at java.base/java.lang.Thread.run(Thread.java:829)
```

### PR DESCRIPTION
**Pull request by laglangyue**
close #113 

I search the changes in mssql release
https://github.com/microsoft/mssql-jdbc/releases?q=encrypt&expanded=true
look at this picture. 
image

## How to solve this for our test
(1) add `encrypt=true;trustServerCertificate=true"` to jdbc url
(2) or add `encrypt=false` to jdbc url
## What can users do?
users should ensure that the sqlserver enable tls,maybe need some config, so that client can connect to sqlserver with TLS

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,4 @@ target
 *.code-workspace
 .bloop
 .metals
+dumps
```
Comment (laglangyue) [inline .gitignore:15]: when I test in local ,this folder generate, I think adding it to ignore  file would be better

**Comment by laglangyue** [comment]
image

I have test it locally, it looks good

**Comment by laglangyue** [comment]
@pjfanning PTAL

**Review by pjfanning — APPROVED** [review]
lgtm - thanks

**Pull request merged**